### DISCUSSION TITLE
when search word order change，i hope the search result will not change

### DISCUSSION THREAD

**Discussion by BrynnChu** [meilisearch/product#370] [Feedback & Feature Proposal]
I have two articles. One article contains the keyword “A” and the other contains the keyword “B”. When I search for “A B”, I will only search for article A. when I search for “B A”, I will only search for article B. I hope that no matter how the keyword order changes, the search results will be the same. How to deal with them?

**Comment by gmourier** [comment]
Hello @chu-bin 👋

It's now possible to select the matching strategy at search time.

What you are describing is the default strategy `last`.

We introduced an `all` strategy that will search for documents containing `A B` and `B A`

📚 https://docs.meilisearch.com/reference/api/search.html#matching-strategy

I'm locking that discussion. Feel free to open a new one if it's needed!
